Explorer/Assets/DCL/LOD/Pools/LODAsset.cs

```diff
@@ -1,100 +1,145 @@
 using DCL.Profiling;
 using ECS.StreamableLoading.AssetBundles;
 using System;
+using Cysharp.Threading.Tasks;
 using DCL.AvatarRendering.AvatarShape.Rendering.TextureArray;
 using DCL.Optimization.Pools;
+using ECS.Unity.SceneBoundsChecker;
+using JetBrains.Annotations;
 using UnityEngine;
+using UnityEngine.Networking;
 using Utility;
+using Object = UnityEngine.Object;
 
 namespace DCL.LOD
 {
-    public struct LODAsset : IDisposable
+    public class LODAsset : IDisposable
     {
-
+        public LOD_STATE State;
         public readonly LODKey LodKey;
-        public readonly GameObject Root;
-        public AssetBundleData AssetBundleReference;
-        public readonly bool LoadingFailed;
+
+        public GameObject Root;
+        public AsyncInstantiateOperation<GameObject> AsyncInstantiation;
+        public  TextureArraySlot?[] Slots;
+        
         private readonly ILODAssetsPool Pool;
-        private readonly TextureArraySlot?[] Slots;
+        internal AssetBundleData AssetBundleReference;
 
-        public LODAsset(LODKey lodKey, GameObject root, AssetBundleData assetBundleReference, ILODAssetsPool pool, TextureArraySlot?[] slots)
-        {
-            LodKey = lodKey;
-            Root = root;
-            AssetBundleReference = assetBundleReference;
-            LoadingFailed = false;
-            Pool = pool;
-            Slots = slots;
+        private readonly string SceneID;
+        private readonly Vector2Int ParcelCoord;
+        private readonly TextureArrayContainer LodTextureArrayContainer;
 
-            ProfilingCounters.LODAssetAmount.Value++;
-            ProfilingCounters.LOD_Per_Level_Values[LodKey.Level].Value++;
+        public enum LOD_STATE
+        {
+            UNINTIALIZED,
+            FAILED,
+            SUCCESS,
+            WAITING_INSTANTIATION,
         }
 
-        //Constructor for LOD_0 which uses the default Scene Material (No texture array)
-        public LODAsset(LODKey lodKey, GameObject root, AssetBundleData assetBundleReference, ILODAssetsPool pool)
+
+        public LODAsset(LODKey lodKey, ILODAssetsPool pool, AssetBundleData assetBundleData,
+            Transform lodsTransformParent, Vector3 baseParcelPosition,
+            Vector2Int parcelCoord, TextureArrayContainer lodTextureArrayContainer)
         {
             LodKey = lodKey;
-            Root = root;
-            AssetBundleReference = assetBundleReference;
-            LoadingFailed = false;
             Pool = pool;
             Slots = Array.Empty<TextureArraySlot?>();
+            AssetBundleReference = assetBundleData;
+            SceneID = LodKey.Hash;
+            ParcelCoord = parcelCoord;
+            LodTextureArrayContainer = lodTextureArrayContainer;
+
+            //NOTE: Using the count API since the one without count does not parent correctly.
+            AsyncInstantiation =
```

Comment:
This class is overloaded with a mix of data and logic. Please consider moving instantiation logic to the system or extensions

Reply:
Suggestions applied. Moved logic to System